File: src/types/checkpoint.ts
export interface WorkspaceCheckpoint {
  workspaceId: string;
  reference: string;
  shortOid: string;
  createdAt: string;
  subject: string;
}

export interface WorkspaceCheckpointDiff {
  workspaceId: string;
  reference: string;
  diff: string;
}

export interface WorkspaceCheckpointRestorePlan {
  workspaceId: string;
  reference: string;
  currentDirty: boolean;
  changedFileCount: number;
  checkpointFileCount: number;
  warnings: string[];
  steps: string[];
}

export interface WorkspaceCheckpointRestoreResult {
  workspaceId: string;
  reference: string;
  applied: boolean;
  message: string;
}

export interface WorkspaceCheckpointDeleteResult {
  workspaceId: string;
  reference: string;
  deleted: boolean;
  message: string;
}

export interface WorkspaceCheckpointBranchResult {
  workspaceId: string;
  reference: string;
  branch: string;
  created: boolean;
  message: string;
}

File: src-tauri/src/models/checkpoint.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceCheckpoint {
    pub workspace_id: String,
    pub reference: String,
    pub short_oid: String,
    pub created_at: String,
    pub subject: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceCheckpointDiff {
    pub workspace_id: String,
    pub reference: String,
    pub diff: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceCheckpointRestorePlan {
    pub workspace_id: String,
    pub reference: String,
    pub current_dirty: bool,
    pub changed_file_count: usize,
    pub checkpoint_file_count: usize,
    pub warnings: Vec<String>,
    pub steps: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceCheckpointRestoreResult {
    pub workspace_id: String,
    pub reference: String,
    pub applied: bool,
    pub message: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceCheckpointDeleteResult {
    pub workspace_id: String,
    pub reference: String,
    pub deleted: bool,
    pub message: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceCheckpointBranchResult {
    pub workspace_id: String,
    pub reference: String,
    pub branch: String,
    pub created: bool,
    pub message: String,
}
